compiler/src/iree/compiler/Codegen/LLVMCPU/TileSizeSelection.h

```diff
@@ -0,0 +1,123 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_COMPILER_CODEGEN_LLVMCPU_TILESIZESELECTION_H_
+#define IREE_COMPILER_CODEGEN_LLVMCPU_TILESIZESELECTION_H_
+
+#include "iree/compiler/Codegen/Dialect/LoweringConfig.h"
+#include "mlir/IR/BuiltinOps.h"
+
+namespace mlir {
+namespace iree_compiler {
+
+/// Provides unified API to get access to all the tile size needed during the
+/// CPU lowering process, while abstracting the representation and verification
+/// details of such information in the IR.
+class TilingConfig {
+ public:
+  TilingConfig() {}
```

Comment:
should `tilingLevelToActualLevelMap` be initialized to `InvalidLevel` in this case?

Reply:
Sorry, I left code uncommitted. The last commit didn't even build. I committed everything and addressed your comments, in case you want to take another quick look. Thanks!

Reply:
Unfortunately, this doesn't work as expected, because we create an invalid `TilingConfig` (without a valid `LowerginConfig`) and then we try to retrieve information from it using the invalid `LoweringConfig`. I think we shouldn't allow to create an invalid `TilingLowering`. I can go back to the previous state and move the repetitive code to a utility function.